api/src/controllers/scrapper/index.ts

```diff
@@ -1,7 +1,182 @@
+  const browser = await puppeteer.launch({ headless: 'shell' })
+  const page = await browser.newPage()
+
+  // Navigate the page to a URL.
+  await page.goto('https://developer.chrome.com/')
+
+  // Set screen size.
+  await page.setViewport({ width: 1080, height: 1024 })
+
+  // Type into search box.
+  await page.locator('.devsite-search-field').fill('automate beyond recorder')
+
+  // Wait and click on first result.
+  await page.locator('.devsite-result-item-link').click()
+
+  // Locate the full title with a unique string.
+  const textSelector = await page
+    .locator('text/Customize and automate')
+    .waitHandle()
+  const fullTitle = await textSelector?.evaluate(el => el.textContent)
+
+  // Print the full title.
+  console.log('The title of this blog post is "%s".', fullTitle)
+
+  const adScreenshot = await page.screenshot({
+    path: 'hn.png',
+  })
+
+  await browser.close()
+}
+
+const demoScreenshot = async () => {
+  // Create a browser instance
+  const browser = await puppeteer.launch()
+
+  // Create a new page
+  const page = await browser.newPage()
+
+  // Set viewport width and height
+  await page.setViewport({ width: 1280, height: 720 })
+
+  const website_url =
+    'https://www.bannerbear.com/blog/how-to-convert-html-into-pdf-with-node-js-and-puppeteer/'
+
+  // Open URL in current page
+  await page.goto(website_url, { waitUntil: 'networkidle0' })
+
+  const filePath = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'generated',
+    'screenshots',
+    'firstScreenshot.png'
+  )
+  await page.screenshot({
+    path: filePath,
+  })
+
+  //   const filePath = path.resolve('@generated/screenshots', 'screenshot.png')
+  //   await page.screenshot({ path: filePath })
+
+  console.log('Finaliza')
+  await browser.close()
+}
+
+const authenticateToSite = async ({ page }: { page: Page }) => {
+  const loginBtn = await page.locator('a.login-btn')
+
+  await Promise.all([
+    loginBtn.click(),
+    page.waitForNavigation({ waitUntil: 'networkidle0' }),
+  ])
+
+  await page.screenshot({
+    path: `ss/login.png`,
+  })
+
+  const emailInput = await page.locator('#email')
+  await emailInput.fill(`${process.env.SEMINUEVOS_USER}`)
+  const passwordInput = await page.locator('#password')
+  await passwordInput.fill(`${process.env.SEMINUEVOS_PASSWORD}`)
+  const submitBtn = await page.locator('button[type="submit"]')
+
+  await page.screenshot({
+    path: `ss/loginForm.png`,
+  })
+
+  await Promise.all([
+    await submitBtn.click(),
+    page.waitForNavigation({ waitUntil: 'networkidle0' }),
+  ])
+
+  await page.screenshot({
+    path: `ss/home.png`,
+  })
+
+  const vendeTuVehiculoBtn = await page.locator('a.btn-primary')
+
+  await Promise.all([
+    await vendeTuVehiculoBtn.click(),
+    page.waitForNavigation({ waitUntil: 'networkidle0' }),
+  ])
+
+  await page.screenshot({
+    path: `ss/vende_tu_vehiculo.png`,
+  })
+}
+
+const browserManager = () => {
```

Comment:
Closure function to handle Puppeteer's initialize browser, set page and close browser processes. This logic should be move into a service file, will do that on following PRs to clean up the controller